src/parcsr_ls/par_amg_setup.c

```diff
@@ -1307,33 +1341,51 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
 
          /**************************************************/
          /*********Set the fixed index to CF_marker*********/
-         //num_C_point_coarse
 
-         if (hypre_ParAMGDataCPointKeepLevel(amg_data) > 0)
+         /* Set fine points (F_PT) given by the user */
+         if ((num_F_points > 0) && (level == 0))
+         {
+            for (j = 0; j < num_F_points; j++)
+            {
+               row = (HYPRE_Int) (F_points_marker[j] - first_local_row);
+               if ((row >= 0) && (row < local_size))
+               {
+                  CF_marker[row] = -1; // Assumes F_PT == -1
+               }
+            }
+         }
+
+         if (hypre_ParAMGDataCPointsKeepLevel(amg_data) > 0)
          {
             if (block_mode)
             {
-               hypre_printf("Keeping coarse nodes in block mode is not implemented\n");
+               if (my_id == 0)
```

Comment:
Most of the time we don't restrict these error messages to process 0.  I think we probably shouldn't ever do that, but I can see the potential downside of getting lots and lots of warnings printed when debugging code.  On the other hand, errors aren't universally going to be the same on all processes and we could, in theory, try to manage the message output in the error routines themselves somehow.  Thoughts?  Comments?

Reply:
I personally find it pretty annoying to get hundreds of error messages and have to parse through for anything else that may be important/don't see any downside to just having one processor print, that's just my two cents though

Reply:
After a bit more thought, I think we should *not* restrict error messages (e.g., with hypre_error() or hypre_error_w_msg()) to a process number.  All of this output goes to stderr and there are a variety of ways to deal with stdio in the mpirun command:

  https://www.open-mpi.org/doc/v3.0/man1/mpirun.1.php

These error messages would only ever be turned on during debugging sessions ('configure --with-print-errors'), and it is better to get the full parallel picture of what's going on when debugging.  From what I can tell, there are only a few other places in the code where we wrap the error messages, and we should remove those if statements when we come across them.

Reply:
In this particular case, block_mode is a variable that assumes the same value across all processors. So, I think it makes sense to throw the error message only from proc 0 and not having redundant messages from other procs as Ben pointed out. Regarding the message output management, I thought about adding a flag to hypre_error_w_msg which decides if the message will be printed by all processors or just the master. Also, in the first case, we could add the processor rank at the beginning of the message, so we would know which rank threw the error when reading it. I saw this in PETSc, by the way.

Reply:
> After a bit more thought, I think we should _not_ restrict error messages (e.g., with hypre_error() or hypre_error_w_msg()) to a process number. All of this output goes to stderr and there are a variety of ways to deal with stdio in the mpirun command:
> 
> https://www.open-mpi.org/doc/v3.0/man1/mpirun.1.php
> 
> These error messages would only ever be turned on during debugging sessions ('configure --with-print-errors'), and it is better to get the full parallel picture of what's going on when debugging. From what I can tell, there are only a few other places in the code where we wrap the error messages, and we should remove those if statements when we come across them.

Thanks for pointing this out, Rob. I was not aware of those options of mpirun. I deleted the "if my_id == 0" statement for that warning message and another one in par_amg_setup. See 5d0ffec